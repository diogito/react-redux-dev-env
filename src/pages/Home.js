import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import logoReact from '../images/logo-react.svg';
import logoRedux from '../images/logo-redux.svg';

import {
  HOME_PAGE_LOADED,
  HOME_PAGE_UNLOADED,
  LOAD_STACK
} from '../constants/actionTypes';

export class Home extends Component {
  stackRow(stackComponent, index) {
    return <div key={index}>{stackComponent.name}</div>;
  }

  UNSAFE_componentWillMount() {
    this.props.onLoad();
    this.props.loadStack();
  }

  componentWillUnmount() {
    this.props.onUnload();
  }

  render() {
    const { stack } = this.props;
    return (
      <div>
        <h1 className="cover-heading">React Redux Development Environment</h1>
        <div className="container">
          <div className="row">
            <div className="col">
              <img src={logoReact} className='App-logo' alt='logo-react' />
            </div>
            <div className="col">
              <img src={logoRedux} className='App-logo' alt='logo-redux' />
            </div>
          </div>
        </div>
        <div>
          { stack && stack.length ?
            <div>{stack.map(this.stackRow)} </div>:
            <div>Loading...</div>
          }
        </div>
      </div>
    )
  }
}

Home.propTypes = {
  stack: PropTypes.array,
  onLoad: PropTypes.func,
  onUnload: PropTypes.func,
  loadStack: PropTypes.func
}

function mapStateToProps({stack}) {
  return {
    stack
  }
}

const mapDispatchToProps = dispatch => ({
  onLoad: () =>
    dispatch({type: HOME_PAGE_LOADED}),
  onUnload: () =>
    dispatch({type: HOME_PAGE_UNLOADED}),
  loadStack: () =>
    dispatch({type: LOAD_STACK})
})

export default connect(mapStateToProps, mapDispatchToProps)(Home);
