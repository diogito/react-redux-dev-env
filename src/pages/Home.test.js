import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from './Home.js';

test('should shallow correctrly', () => {
  const onLoad = jest.fn();
  const loadStack = jest.fn();

  const component = renderer.create(<Home onLoad={onLoad} loadStack={loadStack}/>);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
