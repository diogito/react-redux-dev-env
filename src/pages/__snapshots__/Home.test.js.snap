// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should shallow correctrly 1`] = `
<div>
  <h1
    className="cover-heading"
  >
    React Redux Development Environment
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <img
          alt="logo-react"
          className="App-logo"
          src={[Function]}
        />
      </div>
      <div
        className="col"
      >
        <img
          alt="logo-redux"
          className="App-logo"
          src={[Function]}
        />
      </div>
    </div>
  </div>
  <div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
